be/Controller/UsersCtrl.php

```diff
@@ -99,87 +99,108 @@ public function addUser(Request $request, Response $response): Response
         return $response->withJson([]);
     }
 
-    private function getGame(
-        User $winnerUser,
-        User $looserUser,
-        int $oldWinnerRating,
-        int $oldLooserRating,
-        int $ratingDiff
-    ): Game {
-        $game = new Game();
-        $game->setWinnerUser($winnerUser);
-        $game->setLooserUser($looserUser);
-        $game->setWinnerRatingBefore($oldWinnerRating);
-        $game->setLooserRatingBefore($oldLooserRating);
-        $game->setRatingDiff($ratingDiff);
-
-        return $game;
-    }
+    public function updateRatings(
+        Request $request,
+        Response $response
+    ): Response {
+        $json = $request->getBody();
+        $usersCodes = json_decode($json, true);
+        $winnerUserCode = $usersCodes['winnerUserCode'];
+        $looserUserCode = $usersCodes['looserUserCode'];
 
-    private function calcNewRatings(
-        int $oldWinnerRating,
-        int $oldLooserRating
-    ): array {
-        $kfactor = 32;
+        $this->valdateUpdateRatingsCodes($winnerUserCode, $looserUserCode);
 
-        $winnerLooserDiff = $oldLooserRating - $oldWinnerRating;
-        $pWinner = 1 / (1 + (10 ** ($winnerLooserDiff / 400))); // propability of winning winner user
+        $winnerUser = $this->userRepository->findOneByCode($winnerUserCode);
+        $looserUser = $this->userRepository->findOneByCode($looserUserCode);
 
-        $ratingDiff = round($kfactor * (1 - $pWinner));
-        $newWinnerRating = $oldWinnerRating + $ratingDiff;
-        $newLooserRating = $oldLooserRating - $ratingDiff;
+        $warningMsg = $this->getUpdateUserWarningMsg($winnerUser, $looserUser);
+        if ($warningMsg) {
+            return $response->withJson(['status' => 'warning', 'warningMsg' => $warningMsg]);
+        }
 
-        return [$newWinnerRating, $newLooserRating];
+        $ratingDiff = $this->updateRatingsInDb($winnerUser, $looserUser);
```

Comment:
@kamilkuz 
W tym miejscu PhpStorm mi krzyczy, że encje `$winnerUser`, `$looserUser` mogą być puste (`null`) :/. A wiem, że `getUpdateUserWarningMsg()` to weryfikuje. Nie wiem na razie jak to w sumie zrefaktoryzować, aby było dobrze.
U Ciebie o niczym takim nie informuje?